### PR TITLE
Enable Zip64 support for large files

### PR DESCRIPTION
**Pull request by pkarman**
https://github.com/rubyzip/rubyzip#zip64-support

connects #1132 

## Description

Our Zip config defaults to a max file size of 4G. If there are enough files, or enough large files, compressed in a single .zip, it will error out and return a .zip file that cannot be opened.

This PR fixes that config and adds an optional test to verify. Because the test is slow to run and requires a very large example file, it is off by default and requires the `TEST_LARGE_FILES` env var to run. The intent is that developers should run this locally and not on CI.

```
TEST_LARGE_FILES=true bundle exec rspec spec/services/zipfile_creator_spec.rb
```

*DO NOT* check in the large file that is created. It will over write a local .pdf file. You need to `git checkout lib/pdfs` before committing any changes after running that test.

### CODE REVIEW

File: config/initializers/rubyzip.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# https://github.com/rubyzip/rubyzip#configuration
+
+require "zip"
+
+Zip.setup do |c|
+#  c.on_exists_proc = true
+#  c.continue_on_exists_proc = true
+  c.unicode_names = true
+  c.default_compression = Zlib::BEST_COMPRESSION
+  c.write_zip64_support = true
```
Comment (pkarman) [inline config/initializers/rubyzip.rb]: this is the important line.

File: lib/fakes/document_service.rb [hunk 2]
```diff
@@ -83,11 +83,21 @@ def self.v2_fetch_document_file(record)
...
-      IO.binread(Rails.root + "lib/tiffs/#{record.id % 5}.tiff")
+    filename = if record.mime_type == "application/pdf"
+                 Rails.root + "lib/pdfs/#{record.id % 5}.pdf"
+               else
+                 Rails.root + "lib/tiffs/#{record.id % 5}.tiff"
+               end
+    fsize = File.size(filename)
+    buf = String.new(capacity: fsize) # pre-allocate to reduce number of mallocs
+    # for large files we must read in chunks.
+    # common page size is 4096 (4K) so we want 10k pages of memory (or about 40MB) per chunk.
```
Comment (aroltsch) [inline lib/fakes/document_service.rb]: How did you decide on this number of 40MB per chunk?

Comment (pkarman) [reply lib/fakes/document_service.rb]: Manually tested it with different number of pages to find a size that was fast enough. 4MB was too slow, and anything bigger than 40MB was indistinguishable from 40MB.

File: lib/fakes/document_service.rb [hunk 3]
```diff
@@ -83,11 +83,21 @@ def self.v2_fetch_document_file(record)
   end
 
   def self.file_content(record)
-    if record.mime_type == "application/pdf"
-      IO.binread(Rails.root + "lib/pdfs/#{record.id % 5}.pdf")
-    else
-      IO.binread(Rails.root + "lib/tiffs/#{record.id % 5}.tiff")
+    filename = if record.mime_type == "application/pdf"
```
Comment (aroltsch) [inline lib/fakes/document_service.rb]: I am a little confused because this is in the fakes.

Comment (pkarman) [reply lib/fakes/document_service.rb]: It's in the fakes so we could test the configuration change for zip64.

The actual production (non-fake) code in https://github.com/department-of-veterans-affairs/caseflow-commons/blob/master/app/services/caseflow/s3_service.rb#L40 uses `IO.binread` and it works fine.

In our test we have a single large file (500MB) in order to more quickly create a large .zip file. In production, we have 1000s of smaller files that add up to one large file. So this change to the fakes is to support tests only.

If we find a further error with the `S3Service.fetch_content` we can rewrite in the caseflow-commons repo. But that's outside the scope of this PR (which is mostly just about one line of configuration change).

**Pull request merged**